### PR TITLE
[Streaming API] Improve Streaming API test cases

### PR DESCRIPTION
**Pull request by senthuran16**
## Purpose
- $Subject.
- Adding multi-tenant message context properties to support throttling SSE APIs in tenants (83cf28d). Fixes https://github.com/wso2/product-apim/issues/11179

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+                "          \"subscribe\":{}" +
+                "      }" +
+
+                "   }" +
+                "}";
+        restAPIPublisher.updateAsyncAPI(websocketAPIID, asyncApiDefinition);
+        restAPIPublisher.changeAPILifeCycleStatus(websocketAPIID, APILifeCycleAction.PUBLISH.getAction(), null);
+        waitForAPIDeploymentSync(user.getUserName(), apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
+        createAPIRevisionAndDeployUsingRest(websocketAPIID, restAPIPublisher);
+        Thread.sleep(10000);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java:190]: Any particular reason for this thread sleep?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[] { TestUserMode.SUPER_TENANT_ADMIN },
+                new Object[] { TestUserMode.TENANT_ADMIN }
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java:98]: Don't we need to run this for other user modes? Eg: email user, secondary user store, tenant user etc

Comment (senthuran16) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java:98]: We decided these two are enough for Streaming APIs in an earlier offline discussion with @tharindu1st . Websub[1] and SSE [2] streaming API test cases also have been written in this way.
[1] https://github.com/wso2/product-apim/blob/master/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/websub/WebSubAPITestCase.java#L118
[2] https://github.com/wso2/product-apim/blob/master/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java#L125

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+                "          \"parameters\":{}," +
+                "          \"publish\":{}," +
+                "          \"subscribe\":{}" +
+                "      }" +
+
+                "   }" +
+                "}";
+        restAPIPublisher.updateAsyncAPI(websocketAPIID, asyncApiDefinition);
+        restAPIPublisher.changeAPILifeCycleStatus(websocketAPIID, APILifeCycleAction.PUBLISH.getAction(), null);
+        waitForAPIDeploymentSync(user.getUserName(), apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java:188]: Let's add this line after 189

Comment (senthuran16) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java]: Done in https://github.com/wso2/product-apim/pull/10992/commits/e2ce713e42731bc465c92045d3d537114895b337

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java (hunk 1 above)
Comment (senthuran16) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java]: No particular reason; as discussed offline, this has been removed in https://github.com/wso2/product-apim/pull/10992/commits/e2ce713e42731bc465c92045d3d537114895b337

**Review by mushthaq33 — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java [hunk 4]
```diff
@@ -234,6 +236,17 @@ public void testInvokeSseApi() throws Exception {
         sseReceiver.setReceivedDataEventsCount(0);
     }
 
+    @Test(description = "Invoke SSE API with invalid token", dependsOnMethods = "testInvokeSseApi")
+    public void testInvokeSseApiWithInvalidToken() throws Exception {
+        invokeSseApi("00000000-0000-0000-0000-000000000000", 30000);
+        int received = sseReceiver.getReceivedDataEventsCount();
+        Assert.assertEquals(received, 0);
```
Comment (Nirothipan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java:243]: let's print a message here if it fails with received count ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java [hunk 5]
```diff
@@ -254,81 +267,66 @@ public void testSseApiThrottling() throws Exception {
...
-            @Override
-            public void accept(Boolean aBoolean) {
-                isThrottled.set(aBoolean);
-            }
-        });
-        Thread.sleep(3000L);
-        Assert.assertTrue(isThrottled.get());
+        long connectionAliveDuration = endTime - startTime;
+        Assert.assertTrue(connectionAliveDuration < 60000);
+        Assert.assertTrue(connectionAliveDuration > 0);
```
Comment (Nirothipan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java:329]: here too

Comment (Nirothipan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java:329]: and are we validating throttling based on time, not event counts ?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITopicsTestCase.java [hunk 6]
```diff
@@ -0,0 +1,586 @@
...
+        endpointUri = new URI("http://" + sseServerHost + ":" + sseServerPort);
+
+        apiRequest = new APIRequest(apiName, apiContext, endpointUri, endpointUri);
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.ASYNC_UNLIMITED);
+        apiRequest.setProvider(provider);
+        apiRequest.setType("SSE");
+        HttpResponse addAPIResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId = addAPIResponse.getData();
+        String asyncApiDefinition = "{" +
```
Comment (Nirothipan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITopicsTestCase.java:153]: shall we add this to a resource file and load.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java (hunk 5 above)
Comment (senthuran16) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java]: Done via https://github.com/wso2/product-apim/pull/10992/commits/99f4805af2b1b07ed841e5e5bd000c5178af5384

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java (hunk 4 above)
Comment (senthuran16) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java]: Done via https://github.com/wso2/product-apim/pull/10992/commits/99f4805af2b1b07ed841e5e5bd000c5178af5384

**Review by Nirothipan — APPROVED** [review]

**Pull request closed without merging**